src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+public abstract class Figure implements IAreaCalculator, IRandomSide {
+    private String name;
+    private String color;
+
+    public Figure(String name, String color) {
+        this.name = name;
+        this.color = getColor();
+    }
+
+    public Figure() {
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getColor() {
+        String randomColor = new ColorSupplier().getRandomColor();
+        return randomColor;
+    }
+
+    public void getDrawing() {
+        System.out.println("Figure: " + getName() + " , "
+                + "color: " + getColor() + ".");
+    }
+
+    public void getDefaultDrawing() {
+        System.out.println("Figure: " + getName() + " , "
+                + "area: " + String.format("%.2f", getArea()) + " sq.units, "
+                + "radius: 10 units, "
+                + "color: white.");
+    }

```

Comment:
remove this method from here and create an interface Drawable with void draw() method and implement this interface in Figure, so its heirs could implement it